lib/flagger/cloud.rb

```diff
@@ -246,41 +287,141 @@ def identify_entity(entity_json)
-
-        entity = identify_entity(entity_json)
-        flag = @gating_info&.flag(flagger_flag.flag_name)
-        if flag.nil? && !@ingested_flags.include?(flagger_flag.flag_name) then
-          save_flag(flagger_flag.flag_name)
-        end
-        return_value = flag&.public_send(name, entity)
-
-        if name == 'treatment' and !return_value.nil? then
-          exposure = {
-            type: entity.type,
-            id: entity.id,
-            treatment_id: return_value.treatment_id,
-            treatment: return_value.codename
-          }
+    def treatment(flagger_flag, entity_json)
+      stat = FlaggerUtils::Stat.new('duration__get_treatment')
+      stat.start()
 
-          return_value = return_value.codename
+      entity = identify_entity(entity_json)
+      flag = @gating_info&.flag(flagger_flag.flag_name)
+      if flag.nil? && !@ingested_flags.include?(flagger_flag.flag_name) then
+        save_flag(flagger_flag.flag_name)
+      end
+      return_value = flag&.public_send('treatment', entity)
 
-          save_exposure(exposure)
-        end
+      if !return_value then
+        return return_value
+      end
 
-        stat.stop()
-        save_stat(stat)
+      exposure = {
+        flag: flagger_flag.flag_name,
+        type: entity.type,
+        id: entity.id,
+        treatment: return_value[:treatment].codename,
+        method_called: 'get_treatment',
+        eligible: return_value[:eligible]
+      }
+
+      return_value = return_value[:treatment].codename
+
+      save_exposure(exposure)
 
-        return_value
+      stat.stop()
+      save_stat(stat)
+
+      return_value
+    end
+
+
+    def payload(flagger_flag, entity_json)
+      stat = FlaggerUtils::Stat.new('duration__get_payload')
+      stat.start()
+
+      entity = identify_entity(entity_json)
+      flag = @gating_info&.flag(flagger_flag.flag_name)
+      if flag.nil? && !@ingested_flags.include?(flagger_flag.flag_name) then
+        save_flag(flagger_flag.flag_name)
+      end
+      return_value = flag&.public_send('payload', entity)
+
+      if !return_value then
+        return return_value
+      end
+
+      exposure = {
+        flag: flagger_flag.flag_name,
+        type: entity.type,
+        id: entity.id,
+        treatment: return_value[:treatment].codename,
+        method_called: 'get_payload',
+        eligible: return_value[:eligible]
+      }
+
+      return_value = return_value[:treatment].payload
+
+      save_exposure(exposure)
+
+      stat.stop()
+      save_stat(stat)
+
+      return_value
+    end
+
+    def eligible?(flagger_flag, entity_json)
+      stat = FlaggerUtils::Stat.new('duration__is_eligible')
+      stat.start()
+
+      entity = identify_entity(entity_json)
+      flag = @gating_info&.flag(flagger_flag.flag_name)
+      if flag.nil? && !@ingested_flags.include?(flagger_flag.flag_name) then
+        save_flag(flagger_flag.flag_name)
+      end
+      return_value = flag&.public_send('eligible?', entity)
```

Comment:
We don't need to use `public_send` if the method name is static.
```suggestion
      return_value = flag&.eligible?(entity)
```